File: QTD.SPA/src/app/_DtoModels/Position_Task/LinkR5UpdateTasksModel.ts
import { Position_HistoryCreateOptions } from "../PositionHistory/PositionHistoryCreateOptions";

export class LinkR5UpdateTasksModel {
  link_TaskIds: Array<string>;
  unlinkAll: boolean;
  position_HistoryCreateOptions: Position_HistoryCreateOptions;
  

  setTasksIdsToLink = (id: string) => {
    if (!this.link_TaskIds) this.link_TaskIds = [];
    let taskId = this.link_TaskIds.filter(dd => dd == id)[0];
    if (!taskId) {
      this.link_TaskIds.push(id);
    }
  }
}

File: QTD2.Infrastructure/Model/PositionTask/LinkR5UpdateTasksModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using QTD2.Infrastructure.Model.PositionHistory;

namespace QTD2.Infrastructure.Model.PositionTask
{
    public class LinkR5UpdateTasksModel
    {
        public List<int> Link_TaskIds { get; set; }
        public bool UnlinkAll { get; set; }
        public Position_HistoryCreateOptions Position_HistoryCreateOptions { get; set; }
    }
}